### PR TITLE
[JAX] Allow enabling partial custom calls through the environment variable

### PR DESCRIPTION
**Pull request by zlsh80826**
# Description

This PR is to add an environment variable `NVTE_CUSTOM_CALLS_RE` to enable custom calls. (Default to all)
This is useful for comparing the performance difference between custom calls and XLA.

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refractor

## Changes

- Add `NVTE_CUSTOM_CALLS_RE` to select the custom calls enable sets
- Add XLA implementation for each custom call

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: transformer_engine/jax/cpp_extensions/activation.py [hunk 1]
```diff
@@ -155,6 +189,9 @@ def act_lu(inputs: jnp.ndarray, activation_type: Sequence[Union[str, Callable]])
     Input shape: (N, 1, H) for non-gated activations
                  (N, 2, H) for gated activations
     """
+    if not ActLuPrimitive.enabled():
+        return _act_lu(inputs, activation_type)
```
Comment (phu0ngng) [inline transformer_engine/jax/cpp_extensions/activation.py:193]: I think it is better to rename this function to `jax_act_lu()`

File: transformer_engine/jax/cpp_extensions/normalization.py [hunk 2]
```diff
@@ -239,12 +241,77 @@ def partition(zero_centered_gamma, epsilon, mesh, arg_infos, result_infos):
 register_primitive(LayerNormFwdPrimitive)
 
 
+def _layernorm(x, gamma, beta, zero_centered_gamma, eps):
```
Comment (phu0ngng) [inline transformer_engine/jax/cpp_extensions/normalization.py:244]: If we have all "native jax" implementations for these functions here which are equivalent to reference functions in the unit test, I think we should update unit tests to import and use these functions directly.

File: transformer_engine/jax/cpp_extensions/activation.py [hunk 3]
```diff
@@ -22,6 +23,7 @@
     jax_dtype_to_ir_dtype,
     get_padded_spec,
 )
```
Comment (phu0ngng) [inline transformer_engine/jax/cpp_extensions/activation.py:25]:
```suggestion
)
from flax import linen as nn
```
For the change below

**Review by denera — APPROVED** [review]
LGTM, pending rebase and retest.

File: transformer_engine/jax/cpp_extensions/normalization.py (hunk 2 above)
Comment (zlsh80826) [reply transformer_engine/jax/cpp_extensions/normalization.py:244]: I would like to maintain two separate implementations for both the exposed APIs and unit tests. Otherwise, the reference for the ‘customcalls disable’ path will be self-referential.

File: transformer_engine/jax/cpp_extensions/activation.py [hunk 4]
```diff
@@ -42,6 +44,38 @@
...
+        case NVTE_Activation_Type.SREGLU:
+            activation, linear = activation_type
+            assert linear == "linear"
+            act_input, linear_input = jnp.split(inputs, [1], axis=-2)
+            output = jnp.where(act_input > 0, act_input * act_input, 0) * linear_input
+        case _:
+            raise ValueError(f"Unregonized {activation_type=}")
+    return jnp.squeeze(output, axis=-2)
+
+
```
[the remark this answers is not in the document]
Comment (zlsh80826) [reply transformer_engine/jax/cpp_extensions/activation.py]: done

File: transformer_engine/jax/cpp_extensions/activation.py (hunk 1 above)
Comment (zlsh80826) [reply transformer_engine/jax/cpp_extensions/activation.py]: Named as `_jax_act_lu` and also other functions.

File: tests/jax/test_custom_call_compute.py [hunk 5]
```diff
@@ -19,8 +19,10 @@
 from transformer_engine.jax.fp8 import FP8MetaPackage, FP8Helper, is_fp8_available
 from transformer_engine.jax.layernorm import layernorm, layernorm_fp8_dot
 from transformer_engine.jax.layernorm_mlp import activation_lu, fused_layernorm_fp8_mlp
+from transformer_engine.jax.cpp_extensions.activation import _convert_to_activation_function
```
Comment (phu0ngng) [inline tests/jax/test_custom_call_compute.py:22]: Hi, how about importing and using `_jax_act_lu` instead of `_convert_to_activation_function`?

Comment (zlsh80826) [reply tests/jax/test_custom_call_compute.py]: done

**Review by phu0ngng — APPROVED** [review]
